src/ansi-c/c_typecheck_code.cpp

```diff
@@ -985,6 +985,87 @@ void c_typecheck_baset::typecheck_spec_assigns_target(exprt &target)
   }
 }
 
+void c_typecheck_baset::typecheck_spec_function_pointer_obeys_contract(
+  exprt &expr)
+{
+  if(!can_cast_expr<function_pointer_obeys_contract_exprt>(expr))
+  {
+    error().source_location = expr.source_location();
+    error() << "expected ID_function_pointer_obeys_contract expression in "
+               "requires_contract/ensures_contract clause, found "
+            << id2string(expr.id()) << eom;
```

Comment:
Nit pick: `id2string` is not needed here, `operator<<` will take care.

Reply:
fixed